sql/catalyst/src/test/scala/org/apache/spark/sql/connector/InMemoryTable.scala

```diff
@@ -158,6 +158,42 @@ class InMemoryTable(
   override def deleteWhere(filters: Array[Filter]): Unit = dataMap.synchronized {
     dataMap --= InMemoryTable.filtersToKeys(dataMap.keys, partFieldNames, filters)
   }
+
+
+  // NOTE: this only support updating data with a literal, i.e., it cannot update
+  // a date `d` to `date_add(d, 1)`, which is an function.
+  override def updateWhere(sets: util.Map[String, Expression],
+      filters: Array[Filter]): Unit = dataMap.synchronized {
+    val keys = InMemoryTable.filtersToKeys(dataMap.keys, partFieldNames, filters)
+
+    val newData = keys.map {
+      key =>
+        val newRows = new BufferedRows
+        dataMap.get(key) match {
+          case Some(rows) =>
+            rows.rows.foreach {
+              row =>
+                val values = mutable.Buffer[Any]()
+                row.toSeq(schema).copyToBuffer(values)
```

Comment:
I think we can simply call `toSeq(...).toArray`.

Reply:
Here I want a mutable container which I can update the element at runtime.

Reply:
isn't array mutable?

Reply:
Sorry, my mistake. Updated.